### DISCUSSION TITLE
Enforce project to as end of chain

### DISCUSSION THREAD

**Discussion by pdevito3** [LuckyPennySoftware/AutoMapper#3909] [Ideas]
So @jbogard, we had a fun bug today due to a FirstOrDefault after a ProjectTo. We eventually figured it out & saw the docs call out that this is a bad idea (and it makes sense), but I wonder if there’s a way to have automapper prevent that from being done in the first place.

Ideally there could be a safeguard built into the lib to make sure it's at the end of a chain (and maybe even that it's an entity), but not sure what that would look like. A Roslyn analyzer could work, but won't do much good if people don't install it. 

I'd be down to take a crack at a PR here, but not sure where to start.

**Comment by jbogard** [comment]
No idea because all we do is create a LINQ Select call, there's not really good places for "hooks" for additional behavior. It would have to be an analyzer I think?

**Reply by pdevito3** [reply]
Yeah, that's what I figured 😕 

Is what it is, just figured I'd check in case I could help someone else out down the line!
